CMakeLists.txt

```diff
@@ -67,6 +68,10 @@ if(USE_SBRK)
   add_definitions(-DUSE_SBRK)
 endif()
 
+if(EXPOSE_EXTERNAL_ALLOCATOR_API)
+  add_definitions(-DSNMALLOC_EXPOSE_RESERVE -DSNMALLOC_EXPOSE_PAGEMAP)
```

Comment:
This shouldn't be add_definitions, we should be adding these definitions only for the shim library target, not unconditionally for everything.  We should probably also hide this option on Windows, where the shim library isn't built.